HEYUI-60 Implementation Font package

hcafaq opened the pull request:
## Pull Request

### Description
Adds heycar fonts as css fonts face
Adds heycar fonts as css in js

Describe your changes (or add ticket description) and add relevant screenshots

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change

### How do I test this

run `npm run build`
go to dist folder inside fonts package run `npm link`,
in es-web-app run  `npm link @heycar-uikit/fonts`
in _app.js import like this `import '@heycar-uikit/fonts'` 
run `npm run dev` and see the fonts loaded in the browser

## Checklist

### Did you remember to take care of the following?

- [x] `npm i` – for new NPM dependencies.
- [ ] `npm run lint` - to check for linting issues
- [ ] `npm run test` - to run unit tests
- [ ] `npm run test:screenshots` - to run snapshot tests

### New Feature / Bug Fix

- [ ] Run unit tests to ensure all existing tests are still passing.
- [ ] Add new passing unit tests to cover the code introduced by your pr.

Thanks for contributing!

hcafaq committed:
HEYUI-60 Implementation Font package

derek-heycar commented on packages/fonts/CHANGELOG.md:
```diff
@@ -0,0 +1,12 @@
+# Change Log
```
copy/past changelog

derek-heycar commented on packages/fonts/package.json:
```diff
@@ -0,0 +1,37 @@
+{
+  "name": "@heycar-uikit/fonts",
+  "version": "1.0.3",
```
why is your package already 1.0.3 version ?

derek-heycar commented on packages/fonts/package.json:
```diff
@@ -0,0 +1,37 @@
+    "react": "^17.0.1"
+  },
+  "dependencies": {
+    "classnames": "^2.3.1"
```
do you really need it ?

derek-heycar commented on packages/fonts/src/Fonts.tsx:
```diff
@@ -0,0 +1,63 @@
+  }
+`;
+
+export default fonts;
```
it is not a component, it is just a font that you can import and use inside the market.

derek-heycar commented on packages/fonts/docs/Description.mdx:
```diff
@@ -0,0 +1,6 @@
+
+## Fonts
+
+Use font-face to load fonts
```
needs more description

derek-heycar commented on packages/fonts/docs/Component.stories.mdx:
```diff
@@ -0,0 +1,44 @@
+import { Meta, Story, Props } from '@storybook/addon-docs';
```
it is not needs

derek-heycar commented on packages/fonts/README.md:
````diff
@@ -0,0 +1,21 @@
+npm install @heycar-uikit/fonts
+```
+
+## Usage
+
+```tsx
+import fonts from '@heycar-uikit/fonts';
+```
+
````
how to use it?

hcafaq committed:
copy/past changelog

hcafaq committed:
update package version

hcafaq committed:
remove unneeded dependency

hcafaq committed:
removed needless stories and files

hcafaq committed:
update readme with usage of fonts

hcafaq committed:
lock file

hcafaq committed:
update file structure

derek-heycar reviewed (changes requested):
1. Could you please rename the font files themselves to human-friendly names.
2. use another font format by default, because the ttf it's not a good performance font format it can be a fallback, but not a default. better woff2, woff.
3. and let's divide the font packages by the weight. It is available to import not all lists of the fonts, but you can import fonts by weight 
`@heycar-uikit/fonts` - it will import all list of the font.
`@heycar-uikit/fonts/[200|300|400]`  will be imported only in particular weight of the font

derek-heycar commented on packages/fonts/README.md:
````diff
@@ -0,0 +1,27 @@
+```
+
+## Usage
+
+```tsx
+import '@heycar-uikit/fonts';
+
+p {
+     font-family: 'Objektiv';
+}
+
+h1 {
+     font-family: 'Objektiv';
+}
+```
````
Is it a strange example, is not it?
Do you import the empty javascript file in the javascript and after that you write the CSS into the javascript file?

hcafaq committed:
add files wih seperate weights

hcafaq committed:
Merge branch 'main' of https://github.com/hey-car/hey-ui into feature/HEYUI-60-implementation-font-package

hcafaq committed:
rename font files to user friendly names

hcafaq committed:
import default css

derek-heycar commented on packages/fonts/CHANGELOG.md:
```diff
@@ -0,0 +1,11 @@
+# 1.1.0 (2022-06-15)
+
+### Features
+
+- **button:** button component first implementation ([#28](https://github.com/hey-car/heycar-uikit/issues/28)) ([8b267fb](https://github.com/hey-car/heycar-uikit/commit/8b267fb5cf93cac5cab60889bba036b541349785))
+- **typography:** typography component ([#34](https://github.com/hey-car/heycar-uikit/issues/34)) ([48aa261](https://github.com/hey-car/heycar-uikit/commit/48aa26142d3a0406c732bbe7e8624494172828b7))
```
it is not needed for fonts, it is a changelog from the core package.

derek-heycar commented on packages/fonts/package.json:
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "@heycar-uikit/fonts",
+  "version": "1.0.0",
+  "description": "HeyCar-UIKit fonts",
+  "main": "dist/index.js",
+  "module": "./dist/esm/index.js",
```
how I would be doing it.
set here
```
"main": "path/to/default.css",
```
remove  `"module": "./dist/esm/index.js",` - it is not needed absolutely.

remove bold.css and normal.css, move 200|300|500.css to the root.

add the documentation to the storybook without sandbox.

that's all.

hcafaq committed:
move weight files in root, update package entry

hcafaq committed:
update unicode ranges for font-face

hcafaq committed:
update change log

hcafaq committed:
update story docs

hcafaq committed:
change log out of src folder

hcafaq committed:
update docs with figma design and description

hcafaq committed:
update docs to add _app.tsx reference

derek-heycar reviewed (changes requested):
I remembered so important thing, please replace the fonts in the storybook theme on the font from the package

hcafaq committed:
updating fonts in storybook

derek-heycar committed:
fix small mistake in the docs part

derek-heycar committed:
Merge branch 'main' into feature/HEYUI-60-implementation-font-package

derek-heycar committed:
chore: publish

 - @heycar-uikit/button@1.0.5
 - @heycar-uikit/fonts@1.0.0
 - @heycar-uikit/themes@1.0.6
 - @heycar-uikit/typography@1.0.3
 - @heycar-uikit/vars@1.0.5

derek-heycar committed:
replace path to the entry file

derek-heycar committed:
chore: publish

 - @heycar-uikit/fonts@1.0.1

derek-heycar committed:
fix build for fronts package

derek-heycar committed:
chore: publish

 - @heycar-uikit/fonts@1.0.2

derek-heycar committed:
update type file for the build copy

derek-heycar committed:
chore: publish

 - @heycar-uikit/fonts@1.0.3

derek-heycar committed:
fix(icons): fixed build Fixed build config for the fonts files (woff, woff2)

derek-heycar committed:
chore: publish

 - @heycar-uikit/fonts@1.0.4

derek-heycar commented:
:tada: This PR is included in version 1.3.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@heycar-uikit/core/v/1.3.0)
- [GitHub release](https://github.com/hey-car/heycar-uikit/releases/tag/v1.3.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
